src/main/java/com/google/sps/servlets/VerifyRoomServlet.java

```diff
@@ -0,0 +1,95 @@
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+import java.util.HashMap;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.Key;
+
+/** Servlet that takes in the user's room Id and verifies that it
+ * exists, then prints the json version of all the urls in playlist
+*/
+@WebServlet("/verify-room")
+public final class VerifyRoomServlet extends HttpServlet {
+  private String inputtedUserTag = "user-party-link";
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    response.setContentType("application/json");
+		
+    String currentRoomId = request.getParameter("roomId");
+    // TODO: handle if they inputted a key string that does not exist in datastore
+    Key currentRoomKey = getKeyFromString(currentRoomId);
+    Room currentRoom = Room.fromKey(currentRoomKey);
+
+    // If the user sent in a room id not in the datastore, send them a hardcoded youtube video
+    // TODO: Redirect to a specific page telling the client that they inputted the wrong room id
+    if(currentRoom == null){
+      String jsonOfTempUrl = new Gson().toJson("https://www.youtube.com/embed/Bey4XXJAqS8");
+      response.getWriter().println(jsonOfTempUrl);
+    } else {
+      Queue<Video> videosOfPlaylist = currentRoom.getVideos();
+      ArrayList<String> urlsOfPlaylist = extractVideoUrls(videosOfPlaylist);
+      String jsonOfUrls = new Gson().toJson(urlsOfPlaylist);
+      response.getWriter().println(jsonOfUrls);
+    }
+  }
+
+  /*
+  * Finds the key from the datastore using the String of the ID
+  */
+  public Key getKeyFromString(String roomId) {
+    Query query = new Query(Room.ROOM_ENTITY);
+    PreparedQuery results = datastore.prepare(query);
+    for(Entity currentRoomEntity : results.asIterable()) {
+      Key currentRoomKey = currentRoomEntity.getKey();
+      String parsedRoomKey = currentRoomKey.toString().substring(5,currentRoomKey.toString().length() - 1);
+      if(roomId.equals(parsedRoomKey)){
+        return currentRoomKey;
+      }
+    }
+    return null;
+  }
+
+  /*
+  * Take the queue of videos associated with the room and transfer it into an array
+  */
+  public ArrayList<String> extractVideoUrls(Queue<Video> videosOfPlaylist){
+    ArrayList<String> videoUrls = new ArrayList<>();
+    while(!videosOfPlaylist.isEmpty()){
+      Video currVideo = videosOfPlaylist.remove();
+      videoUrls.add(currVideo.getUrl());
+    }
+    return videoUrls;
+  }
```

Comment:
videosOfPlaylist.stream().map(Video::getUrl).collect(Collectors.toList());

Also, don't delete from the direct copy of the Video Queue. This will cause issues (because you lose that video when a new user joins).